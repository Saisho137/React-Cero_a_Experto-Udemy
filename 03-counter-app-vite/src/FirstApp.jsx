
//This object will be written to memory only once.
const object = {
    name: 'Santiago',
    lastName: 'Betancur'
};

const App = () => {
    //Put your data, objects or methods here if depends on component state/operations.
    //This functions will be written to memory after each re render.
    const printHelloWorld = () => 'Hello world!';

    return (
        <>
            <h1>{`${object.name} ${object.lastName}`}</h1>
            <h2>{printHelloWorld()}</h2>
            <p>Description</p>
        </>
    )
}

export default App;