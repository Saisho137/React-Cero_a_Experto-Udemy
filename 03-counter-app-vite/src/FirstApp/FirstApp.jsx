import PropTypes from 'prop-types';

//This object will be written to memory only once.
const object = {
    name: 'Santiago',
    lastName: 'Betancur'
};

const FirstApp = ({title, bestNumber = 123, noParentProp}) => {
    //Put your data, objects or methods here if depends on component state/operations.
    //This functions will be written to memory after each re render.
    const printHelloWorld = () => 'Hello world!';

    return (
        <>
            <h1>{title}</h1>
            <h2>{printHelloWorld()}</h2>
            <p>{`${object.name} ${object.lastName}` + noParentProp}</p>
            <h3>{bestNumber / 10}</h3>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    bestNumber: PropTypes.number.isRequired,
}

FirstApp.defaultProps = {
    title: 'No hay titulo',
    bestNumber: 321,
    noParentProp: ' - Not from parent'
}

export default FirstApp;