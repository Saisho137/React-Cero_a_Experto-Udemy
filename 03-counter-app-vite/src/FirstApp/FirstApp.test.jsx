
import { render } from '@testing-library/react';
import FirstApp from './FirstApp';

describe('Pruebas en <FirstApp />', () => {
    test('Debe hacer match con el snapshot', () => {
        const title = 'Hola mundo';
        const { container } = render(<FirstApp title={title} />);

        expect(container).toMatchSnapshot();
    });
    test('Debe mostrar el titulo en h1', () => {
        const title = 'Hola mundo';
        const { container, getByText } = render(<FirstApp title={title} />);

        const h1 = container.querySelector('h1');

        expect(h1.innerHTML).toContain(title);
        expect(getByText(title)).toBeTruthy();
    });
})
