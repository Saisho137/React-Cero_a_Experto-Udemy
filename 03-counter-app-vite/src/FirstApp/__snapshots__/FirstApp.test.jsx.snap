// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola mundo
  </h1>
  <h2>
    Hello world!
  </h2>
  <p>
    Santiago Betancur - Not from parent
  </p>
  <h3>
    32.1
  </h3>
</div>
`;
